huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go

```diff
@@ -0,0 +1,119 @@
+package iotda
+
+import (
+	"context"
+	"log"
+	"regexp"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/iotda/v5/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func ResourceAmqp() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAmqpCreate,
+		ReadContext:   resourceAmqpRead,
+		DeleteContext: resourceAmqpDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(8, 128),
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z-_0-9.:]*$`),
+						"Only letters, digits, hyphens (-), underscores (_), dots(.) and colons(:) are allowed"),
+				),
+			},
+		},
+	}
+}
+
+func resourceAmqpCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcIoTdaV5Client(region)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	createOpts := model.AddQueueRequest{
+		Body: &model.QueueInfo{
+			QueueName: d.Get("name").(string),
+		},
+	}
+	log.Printf("[DEBUG] Create IoTDA AMQP params: %#v", createOpts)
```

Comment:
please use AMQP queue to keep same with the doc

Reply:
ok